lib/puppet/provider/cisco_route_map/cisco.rb

```diff
@@ -0,0 +1,588 @@
+                                            ROUTE_MAP_ARRAY_FLAT_PROPS)
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:array_nested, self, '@nu',
+                                            ROUTE_MAP_ARRAY_NESTED_PROPS)
+
+  def initialize(value={})
+    super(value)
+    rmname = @property_hash[:rmname]
+    sequence = @property_hash[:sequence]
+    action = @property_hash[:action]
+    @nu = Cisco::RouteMap.maps[rmname][sequence][action] unless
+      rmname.nil? || sequence.nil? || action.nil?
+    @property_flush = {}
+  end
+
+  def self.properties_get(rmname, sequence, action, nu_obj)
+    debug "Checking route map instance, #{rmname} #{sequence} #{action}"
+    current_state = {
+      name:     "#{rmname} #{sequence} #{action}",
+      rmname:   rmname,
+      sequence: sequence,
+      action:   action,
+      ensure:   :present,
+    }
+
+    # Call node_utils getter for each property
+    (ROUTE_MAP_NON_BOOL_PROPS).each do |prop|
+      current_state[prop] = nu_obj.send(prop)
+    end
+    ROUTE_MAP_ARRAY_FLAT_PROPS.each do |prop|
+      current_state[prop] = nu_obj.send(prop)
+    end
+    ROUTE_MAP_ARRAY_NESTED_PROPS.each do |prop|
+      current_state[prop] = nu_obj.send(prop)
+    end
+    ROUTE_MAP_BOOL_PROPS.each do |prop|
+      val = nu_obj.send(prop)
+      if val.nil?
+        current_state[prop] = nil
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    rm_instances = []
+    Cisco::RouteMap.maps.each do |rmname, sequences|
+      sequences.each do |sequence, actions|
+        actions.each do |action, nu_obj|
+          rm_instances << properties_get(rmname, sequence, action, nu_obj)
+        end
+      end
+    end
+    rm_instances
+  end # self.instances
+
+  def self.prefetch(resources)
+    rm_instances = instances
+    resources.keys.each do |id|
+      provider = rm_instances.find do |rmi|
+        rmi.rmname.to_s == resources[id][:rmname].to_s &&
+        rmi.sequence.to_s == resources[id][:sequence].to_s &&
+        rmi.action.to_s == resources[id][:action].to_s
+      end
+      resources[id].provider = provider unless provider.nil?
+    end
+  end
+
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def properties_set(new_rm=false)
+    ROUTE_MAP_ALL_PROPS.each do |prop|
+      next unless @resource[prop]
+      send("#{prop}=", @resource[prop]) if new_rm
+      unless @property_flush[prop].nil?
+        @nu.send("#{prop}=", @property_flush[prop]) if
+          @nu.respond_to?("#{prop}=")
+      end
+    end
+    # custom setters which require one-shot multi-param setters
+    match_community_set
+    match_ext_community_set
+    match_ipv4_multicast_set
+    match_ipv6_multicast_set
+    match_ip_addr_access_list
```

Comment:
For consistency, please add `_set` to the methods in this list that don't follow that convention.